Bug: reinhardt-test-support missing validator dependency causes CI failure

## Description

\`reinhardt-test-support\` (\`tests/Cargo.toml\`) shares its \`[lib]\` path with \`tests/integration/src/lib.rs\`. The file \`pre_validate.rs\` in that directory uses \`validator::Validate\`, but the \`validator\` crate is only listed as a dependency in \`tests/integration/Cargo.toml\` (\`reinhardt-integration-tests\`), not in \`tests/Cargo.toml\` (\`reinhardt-test-support\`).

This causes \`cargo check --workspace --all-features --tests\` to fail with:
- \`error[E0432]: unresolved import \`validator\`\`
- \`error[E0599]: no method named \`validate\` found\`

## Steps to reproduce

\`\`\`bash
cargo check -p reinhardt-test-support --all-features --tests
\`\`\`

## Expected behavior

Compilation should succeed.

## Fix

Add \`validator = { version = "0.20", features = ["derive"] }\` to \`tests/Cargo.toml\` dependencies.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/tests/Cargo.toml
+++ b/tests/Cargo.toml
@@ -82,6 +82,7 @@ sqlx = { workspace = true }
 reqwest = { version = "0.12", features = ["json"] }
 reinhardt-query = { workspace = true, features = ["derive"] }
 utoipa = { version = "^5.4.0", features = ["url", "uuid", "yaml"] }
+validator = { version = "0.20", features = ["derive"] }
 ctor = "0.6.0"
 linkme = "0.3"
 

